### ISSUE TITLE
Enhancement: Choose own Avatar (persistent)

### ISSUE THREAD

**Issue by demlak** [jitsi/jitsi-meet-electron#207]
Hi,
i would like to choose my own avatar.
I tried to replace avatar url in `~/.config/Jitsi Meet/config.json`, but it will get automaticaly replaced by default random-avatarURL after changing Name or EMail in settings.

**Comment by akshitkrnagpal** [comment]
> I tried to replace avatar url in ~/.config/Jitsi Meet/config.json, but it will get automaticaly replaced by default random-avatarURL after changing Name or EMail in settings.

That is because we generate the avatar URL from email and name. You can see this middleware for the relevant code. https://github.com/jitsi/jitsi-meet-electron/blob/master/app/features/settings/middleware.js

Under the hood, we use Gravatar service so you can add profile pic to gravatar service and use the same email in the app.

**Comment by demlak** [comment]
For privacy ressources we don't want to use gravatar or any other external services.

**Comment by yodaphone** [comment]
I'd like to use my own images as well & not rely on gravatar. any ideas on how to do this?

**Comment by yodaphone** [comment]
> > I tried to replace avatar url in ~/.config/Jitsi Meet/config.json, but it will get automaticaly replaced by default random-avatarURL after changing Name or EMail in settings.
> 
> That is because we generate the avatar URL from email and name. You can see this middleware for the relevant code. https://github.com/jitsi/jitsi-meet-electron/blob/master/app/features/settings/middleware.js
> 
> Under the hood, we use Gravatar service so you can add profile pic to gravatar service and use the same email in the app.

i see that this has been changed to https://abotars.jitsi.net/meeple/

is there a possibility that we can choose an avatar from a local png file instead?

**Comment by saghul** [comment]
Not really. The avatar needs to be a URL (and a data URL won’t do).

I’m losing this since we have no plans to support it, sorry.

**Comment by shdhdieifg** [comment]
I WANT MY AVATR!!!!!!!!!!!!!!

**Comment by dchmelik** [comment]
Please don't force URL/gravatar rather than let users open their own avatar image.

**Comment by func0der** [comment]
I mean, this is not really a solution that is going into the software, but maybe a hack, that can work:

```
const avatarUrl = prompt('What is your avatar url (no data url)?'); const baseSettings = JSON.parse(await window.localStorage.getItem('features/base/settings')); baseSettings.avatarURL = avatarUrl; await window.localStorage.setItem('features/base/settings', JSON.stringify(baseSettings)); window.location.reload();
```

It is manual labour for each user, but it works.

**Comment by saghul** [comment]
That exposes other users to executing whatever code in that URL, it's not a good idea.

**Comment by func0der** [comment]
I added a note for using it. I did not realize that this was the electron repo, but what works in the browser probably will work in a wrapped chromium.

Either way the functionality to load an avatar from that exact key is there. At some point, in some setup, jitsi is using an image url out of that key anyways. If it comes from gravatar or some (trusted) third party url does not really matter. All of them are potentially dangerous. 

I do not say it is the right thing to do. I just wanted to help ppl out that do not want to use an untrusted (judging from the comments) third party provider ;)
